test/pylib/schema_helper.py

```diff
@@ -0,0 +1,238 @@
+                f"AND table_name = '{table.name}'"
+        elif type(table) is str:
+            if table.startswith(f"{self.keyspace}."):
+                table = table[len(self.keyspace) + 1:]
+            tables = {table}
+            q = f"SELECT table_name FROM system_schema.tables WHERE keyspace_name = '{self.keyspace}' " \
+                f"AND table_name = '{table}'"
+        else:
+            tables = set(t.name for t in self.tables)
+            q = f"SELECT table_name FROM system_schema.tables WHERE keyspace_name = '{self.keyspace}'"
+
+        res1 = {row.table_name for row in await self.cql.run_async(q)}
+        assert not tables - res1, f"Tables {tables - res1} not present"
+
+        for table_name in tables:
+            table = next(t for t in self.tables if t.name == table_name)
+            cols = {c.name: c for c in table.columns}
+            c_pos = {c.name: i for i, c in enumerate(table.columns)}
+            res2 = {row.column_name: row for row in await self.cql.run_async(
+                    f"SELECT column_name, position, kind, type FROM system_schema.columns "
+                    f"WHERE keyspace_name = '{self.keyspace}' AND table_name = '{table_name}'")}
+            assert res2.keys() == cols.keys(), f"Column names for {table_name} do not match " \
+                                               f"expected ({', '.join(cols.keys())}) " \
+                                               f"got ({', '.join(res2.keys())})"
+            for c_name, c in res2.items():
+                pos = c_pos[c_name]
+                col = cols[c_name]
+                assert c.type == col.ctype.name, f"Column {c_name} type does not match {c.type} {col.ctype.name}"
+                if pos == 0:
+                    kind = "partition_key"
+                    schema_pos = 0
+                elif pos < table.pks:
+                    kind = "clustering"
+                    schema_pos = 0
+                else:
+                    kind = "regular"
+                    schema_pos = -1
+                assert c.kind == kind, f"Column {c_name} kind does not match {c.kind} {kind}"
+                assert c.position == schema_pos, f"Column {c_name} position {c.position} does not match {schema_pos}"
+
+
+# Until Cassandra 4, NetworkTopologyStrategy did not support the option
+# replication_factor (https://issues.apache.org/jira/browse/CASSANDRA-14303).
+# We want to allow these tests to run on Cassandra 3.* (for the convenience
+# of developers who happen to have it installed), so we'll use the older
+# syntax that needs to specify a DC name explicitly. For this, will figure
+# out the name of the current DC, so it can be used in NetworkTopologyStrategy.
+async def this_dc(cql: CassandraSession):
+    res = await cql.run_async("SELECT data_center FROM system.local")
+    return res[0][0]
+
+
+async def create_keyspace(cql: CassandraSession):
+    keyspace_name = f"k_{new_keyspace_id():02}"
+    dc = await this_dc(cql)
```

Comment:
nitpick: It's a waste to find this_dc(cql) on every create_keyspace from scratch.
It's not important because there is no reason to create many keyspaces in a test anyway.

Reply:
Removed in v3